src/types/markers.js

```diff
@@ -387,6 +387,10 @@ export type FileIoPayload = {|
   source: string,
   operation: string,
   filename: string,
+  // FileIO markers that are happening on the current thread don't have a threadId,
+  // but they have threadId field if the markers belong to a different (potentially
+  // non-profiled) thread.
```

Comment:
Can you explain what happens for profile from before gerald's patch? Were they all on the same thread?

Reply:
Yes, only from the main thread, with the "mainthreadio" feature.
The backend will soon add "fileio" and "fileioall" features, to record IOs from other threads (respectively: all profiled threads, and all threads); See [bug 1464506 "Add off-main thread IO Marker support to Gecko profiles"](https://bugzilla.mozilla.org/show_bug.cgi?id=1464506).

Reply:
Again, agree with Gerald. Added a comment to explain this.